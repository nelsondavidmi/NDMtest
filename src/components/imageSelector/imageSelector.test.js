import React from 'react';
import {shallow} from 'enzyme';
import ImageSelector from './index';

const mockOnCancelImage = jest.fn();
const mockOnChooseImage = jest.fn();

describe('ImageSelector', () => {
  const props = {
    imageURI:
      'https://www.elcarrocolombiano.com/wp-content/uploads/2019/01/20190121-ESTOS-SON-LOS-COLORES-DE-CARRO-M%C3%81S-VENDIDOS-EN-2018-01.jpg',
    imageIsFetching: true,
    label: 'test',
    onCancelImage: mockOnCancelImage,
    onChooseImage: mockOnChooseImage,
    image: '',
  };
  describe('ImageSelector', () => {
    it('should match to snapshot', () => {
      const component = shallow(<ImageSelector {...props} />);
      expect(component).toMatchSnapshot();
    });
    it('should be loading', () => {
      const component = shallow(<ImageSelector {...props} />);
      const activity = component.find('ActivityIndicator');

      expect(activity.length).toBe(1);
    });
    it('Should run onChooseImage', () => {
      const component = shallow(<ImageSelector {...props} />);
      component.setProps({
        imageIsFetching: false,
        imageURI: '',
      });
      const onChooseImage = component.find({testID: 'onChooseImage'});

      onChooseImage.simulate('press');
      expect(mockOnChooseImage).toHaveBeenCalled();
    });
  });
});
