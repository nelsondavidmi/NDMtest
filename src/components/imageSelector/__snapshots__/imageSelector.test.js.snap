// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSelector ImageSelector should match to snapshot 1`] = `ShallowWrapper {}`;
